command/cp.go

```diff
@@ -966,3 +1012,57 @@ func guessContentType(file *os.File) string {
 	}
 	return contentType
 }
+
+type CountingWriter struct {
```

Comment:
How about we merge CountingWrite and CountingReader? such as `countingReadWriter`.